OpenSim/Common/Component.cpp

```diff
@@ -192,6 +192,48 @@ void Component::finalizeFromProperties()
 
     markPropertiesAsSubcomponents();
     componentsFinalizeFromProperties();
+
```

Comment:
Also, could you add a high-level comment describing what this section of code does?

Reply:
Done.